cql/ManageSpecialPopulation.cql

```diff
@@ -291,18 +379,61 @@ define RecommendationForPatientsYoungerThan25:
           '*Refer to ASCCP Consensus Guideline 2019 Section K.1'
         }
       }
-    when ( // Heading 3.1c
+
+    when ( // Heading K.1.3.1 case b
       AgeInYears() < 25 and
+      Rare.CytologyInterpretedAsNilm and  
       Rare.HistologyInterpretedAsCin1OrNormal and
-      Rare.CytologyInterpretedAsAscH and
-      Rare.MostRecentCytologyReport.date less than 12 months before Collate.MostRecentBiopsyReport.date
+      SecondMostRecentCytologyInterpretedAsHsil and
+      Rare.SecondMostRecentCytologyReport.date less than 12 months before Collate.MostRecentBiopsyReport.date and
+      Rare.MostRecentCytologyReport.date less than 3 years after Collate.MostRecentBiopsyReport.date
     ) then
       {
-        short: 'Cytology',
+        short: 'Colposcopy and Cytology',
         date: Collate.DateOfMostRecentReport + 1 year,
         group: 'Younger Than 25 (K.1.3)',
         details: {
-          'Observation is recommended with cytology in 1 year.',
+          'Observation is recommended with colposcopy and cytology in 1 year.',
+          'Diagnostic excisional procedures are not recommended for patients younger than 25 with some exceptions.*',
+          '*Refer to ASCCP Consensus Guideline 2019 Section K.1'
+        }
+      }
+
+    when ( // Heading K.1.3.1 case c
+      // 1 biopsy after CIN1 preceded by HSIL
+      AgeInYears() < 25 and
+      Rare.SecondMostRecentHistologyInterpretedAsCin1OrNormal and
+      Rare.CytologyInterpretedAsHsil and
+      Rare.MostRecentCytologyReport.date less than 12 months before Rare.SecondMostRecentBiopsyReport.date and
+      Rare.HistologyInterpretedAsCin1OrNormal  and
+      Collate.MostRecentBiopsyReport.date less than 3 years after Rare.SecondMostRecentBiopsyReport.date
+          ) then
+      {
+        short: 'Colposcopy and Cytology',
+        date: Collate.DateOfMostRecentReport + 1 year,
+        group: 'Younger Than 25 (K.1.3)',
+        details: {
+          'Observation is recommended with colposcopy and cytology in 1 year.',
+          'Diagnostic excisional procedures are not recommended for patients younger than 25 with some exceptions.*',
+          '*Refer to ASCCP Consensus Guideline 2019 Section K.1'
+        }
+      }
+
+    when ( // Heading K.1.3.1 case d
+      // 1 cyto and 1 biopsy after CIN1 preceded by HSIL
+      AgeInYears() < 25 and
+      Rare.SecondMostRecentHistologyInterpretedAsCin1OrNormal and
+      SecondMostRecentCytologyInterpretedAsHsil and
+      Rare.SecondMostRecentCytologyReport.date less than 12 months before Rare.SecondMostRecentBiopsyReport.date and
+      Collate.MostRecentBiopsyReport.date less than 3 years after Rare.SecondMostRecentBiopsyReport.date and
```

Comment:
```suggestion
      Rare.SecondMostRecentCytologyReport.date less than 12 months before Rare.SecondMostRecentBiopsyReport.date and
      Rare.HistologyInterpretedAsCin1OrNormal and
      Rare.CytologyInterpretedAsNilm and
      Collate.MostRecentBiopsyReport.date less than 3 years after Rare.SecondMostRecentBiopsyReport.date and
```

Don't we need to check for the results of the most recent tests?

Reply:
It looks like this still needs to be addressed.

Reply:
Now checks most recent Histology and cytology in 3 year window after most recent or second most recent biopsy report.